File: RepositoryPattern/MyApplication/app/src/main/java/com/example/myapplication/GsonUtil.java
package com.example.myapplication;

import com.google.gson.GsonBuilder;

public class GsonUtil {
    private static final GsonUtil GSON_UTIL = new GsonUtil();

    public GsonUtil() {
    }

    public static GsonUtil getInstance() {
        return GSON_UTIL;
    }

    public GsonBuilder buildGson() {
        return (new GsonBuilder()).setDateFormat("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'SSS'Z'");
    }
}

File: RepositoryPattern/SampleApplicationForGM/network-util/src/main/java/com/example/networkutil/GsonUtil.kt
package com.example.networkutil

import com.google.gson.GsonBuilder

class GsonUtil {
    fun buildGson(): GsonBuilder {
        return GsonBuilder().setDateFormat("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'SSS'Z'")
    }

    companion object {
        val instance = GsonUtil()
    }
}